### PR TITLE
Add tokenizer and related tests

### PR DESCRIPTION
**Pull request by bharat-biradar**
Adds tokenizer and tests for tokenizer.

### CODE REVIEW

File: lib/src/license_detection/tokenizer.dart [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+  /// Normalized form of the text in [span].
+  final String value;
+
+  /// Denotes the token position.
+  final int index;
+
+  /// Zero based line number of token.
+  int get line => span.start.line;
+
+  /// SourceSpan of the token.
```
Comment (jonasfj) [inline lib/src/license_detection/tokenizer.dart]:
```suggestion
  /// Location of the token in the original license text.
```

File: lib/src/license_detection/tokenizer.dart [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+
+/// A token in this context is a sequence of characters
+/// scanned from the original text separated by a space
+/// or new line.
+@sealed
+class Token {
+  /// Normalized form of the text in [span].
+  final String value;
+
+  /// Denotes the token position.
```
Comment (jonasfj) [inline lib/src/license_detection/tokenizer.dart]: Maybe we can say something like:
```suggestion
  /// Index of the token in the list of tokens from the tokenized license text.
  ///
  /// When a license is tokenized it becomes a list of tokens.
  /// The [index] is the offset of this token within the list of tokens returned from [tokenize] when
  /// the original license text was tokenized.
  ///
  /// Example:
  /// ```dart
  /// final tokens = tokenize('hello world');
  /// assert(tokens[0].index == 0);
  /// assert(tokens[1].index == 1);
  /// ```
```

Yes, yes, it is probably overkill to have an example here... but it's a good exercise in documenting things precisely.

File: test/tokenizer_test.dart [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+void main() {
+  group('Tokenizer tests', () {
+    test('Ignore pure punctuations', () {
+      final text = '// hello! ^& world %^& 1.1';
+      final expected = <String>['hello', 'world', '1.1'];
+      final actual = tokenize(text);
+
+      testOutput(actual, expected);
+    });
```
Comment (jonasfj) [inline test/tokenizer_test.dart]: These tests are really nice!!! 

Consider making a helper function like:
```dart
testTokenizer(String name, {required String text, required List<String> tokens}) => test(name, () {
  testOutput(tokenize(text), tokens);
});

// Then you can just write each test case as:
testTokenize(
  'Ignore pure punctuations',
  text: '// hello! ^& world %^& 1.1',
  tokens: ['hello', 'world', '1.1'],
);
```

Comment (jonasfj) [reply test/tokenizer_test.dart]: Another option is to make a `Map<String,List<String>>` with test vectors and just use the input text as the name of the test case..

This is only really to make easier to write a lot of small test cases. Which is probably a good idea to do :D

File: lib/src/license_detection/tokenizer.dart (hunk 2 above)
Comment (bharat-biradar) [reply lib/src/license_detection/tokenizer.dart]: It's okay to omit example right?

**Pull request merged**